### PR TITLE
Fix firefox youtube issue

### PR DESCRIPTION
**Pull request by ngaro**
- Adds comments on how to run the firefox container
- Fixes crashing tabs in certain cases. This is best visible when playing youtube movies

### CODE REVIEW

**Comment by ngaro** [comment]
I didn't add this fixes to the alpine-version of firefox because it is pretty much broken beyond repair...

**Comment by koep** [comment]
I tested it and it seems stable - ping @jessfraz

File: firefox/Dockerfile [hunk 1]
```diff
@@ -19,6 +30,18 @@ RUN apt-get update && apt-get install -y \
 
 ENV LANG en-US
 
+#Creates a default profile where browser.tabs.remote.autostart is turned off.
+#This fixes tabs crashing in certain cases (e.g. playing youtube movies).
+RUN mkdir -p /root/.mozilla/firefox/abcdefgh.default
+RUN echo 'user_pref("browser.tabs.remote.autostart", false);' > /root/.mozilla/firefox/abcdefgh.default/user.js
```
Comment (koep) [inline firefox/Dockerfile]: Should this maybe be replaced by a COPY instruction (like local.conf)?

File: firefox/Dockerfile [hunk 2]
```diff
@@ -19,6 +30,18 @@ RUN apt-get update && apt-get install -y \
...
+RUN echo '[General]\n\
```
Comment (koep) [inline firefox/Dockerfile]: See above. COPY seems more clean in this case right?

File: firefox/Dockerfile [hunk 3]
```diff
@@ -19,6 +30,18 @@ RUN apt-get update && apt-get install -y \
 
 ENV LANG en-US
 
+#Creates a default profile where browser.tabs.remote.autostart is turned off.
+#This fixes tabs crashing in certain cases (e.g. playing youtube movies).
+RUN mkdir -p /root/.mozilla/firefox/abcdefgh.default
```
Comment (Nathapon8899) [inline firefox/Dockerfile]: Run firefox in a container

**Comment by mviereck** [comment]
I suggest not to create a user profile setting in `/root` but to change `/etc/firefox/syspref.js` instead.

I am running `jess/firefox` not as root but as an unprivileged user in container. Currently the fix does not apply in that case. Changing the setting system wide would apply for all container users.

I suggest a line:
```
 RUN echo 'user_pref("browser.tabs.remote.autostart", false);' >> /etc/firefox/syspref.js
```

**Comment by elerch** [comment]
So I went down the rabbit hole of audio + unprivileged user last night and got pretty much nowhere. @mviereck - do you have any more data on the cause of the issue? This [Nix issue and comment](https://github.com/NixOS/nixpkgs/issues/34822#issuecomment-366540410) describes pretty well what I'm seeing, but the goal I think in this case would be to let FF use PA as provided by the host.

**Comment by ngaro** [comment]
Not that I'm trying to create a heavy discussion (although it will probably follow), but is not running as root worth all the extra issues ?

It's a container after all...

**Comment by mviereck** [comment]
> @mviereck - do you have any more data on the cause of the issue? 

Not really yet - I don't know what `browser.tabs.remote.autostart` does at all. It fixes crashing tabs as @ngaro found out (thank you!).
It also fixes "my" ticket #360. (I found that insecure option `--ipc=host` fixes crashing tabs, too.)

I suggest to transform @ngaro's solution to this short setup that works for all container users:
```
 RUN echo 'pref("browser.tabs.remote.autostart", false);' >> /etc/firefox/syspref.js
```
> So I went down the rabbit hole of audio + unprivileged user last night and got pretty much nowhere. 

The audio thing is a different issue. Maybe we should discuss it in #410. I have a working pulseaudio setup. Sharing `/dev/snd` fails here, too.

**Comment by elerch** [comment]
To add to @mviereck's comment, typically with the browser you're hosting your profile and downloads on the host (you could use a volume, but I don't think that's typical - as if anything anyone does running desktop apps in containers "typical" :) ). In that case, you end up with your home directory with root-owned files that are downloaded from the Internet.

Thanks for the legwork @mviereck ! I'll check out your PR shortly.

**Comment by jessfraz** [comment]
fixed by https://github.com/jessfraz/dockerfiles/pull/429 thanks!

**Pull request closed without merging**